projects/ideabox-groups.md

```diff
@@ -0,0 +1,210 @@
+
+Throughout the project, one of our focuses will be on providing a fluid and responsive client-side
+interface. To this end, we'll rely on JavaScript and jQuery to implement snappy filtering in the
+browser, and `localStorage` to persist our wonderful ideas between sessions.
+
+## Project Requirements
+
+### Architecture
+
+For this project, we'll be increasingly thinking about the "data model" and "dom model" as separate entities. We'll be using:
+
+- JSON and `localStorage` to persist data on page reload.
+- JavaScript to manage client-side interactions.
+
+Your entire application will consist of one HTML page or template.
+You will have two javascript files:
+
+1. An `idea.js` file that contains an `Idea` class. This must be the first script so that your `main.js` file has access to your `Idea` class.
+  * `Idea` methods must include, but are not limited to:
+    1. `constructor`
+    2. `saveToStorage`
+    3. `deleteFromStorage`
+    4. `updateSelf`
+    5. `updateQuality`
+2. A `main.js` file that contains all dom related javascript.
+
+### Data Model
+
+* An idea has an _id_, _title_, a _body_, and a _quality_.
+  * The _id_ should be a unique identifier.
+  * _title_ and _body_ are free-form strings.
+  * _quality_ should be one of the follow: "genius", "plausible", or "swill."
+  * By default, the idea's _quality_ should default to the lowest setting, which is "swill".
+
+Each idea should be created as an instance of the `Idea` class.
+
+### User Flows
+
+#### Viewing ideas
+
+When visiting the application, the user should:
+
+* See a list of all existing ideas, including the title, body, and quality for each idea.
+* Ideas should appear in descending chronological order (with the most recently created idea at the top).
+
+#### Adding a new idea
+
+On the application's main page, a user should:
+
+* See two text boxes for entering the "Title" and "Body" for a new idea,
+  and a "Save" button for committing that idea.
+
+When a user clicks "Save":
+
+* A new idea with the provided title and body should appear in the idea list.
+* The text fields should be cleared and ready to accept a new idea.
+* The page _should not_ reload.
+* The idea should be persisted. It should still be present upon reloading the page.
+
+_Note:_ `localStorage` will not track the type of object, so on page reload you will need to reinstantiate all of your idea instances so that have access to their methods.
+
+#### Deleting an existing idea
+
+When viewing the idea list:
+
+* Each idea in the list should have a link or button to remove it from both the data model and the dom.
+* Upon clicking "Delete", the appropriate idea should be removed from the list.
+* The page _should not_ reload when an idea is deleted.
+* The idea should be removed from `localStorage`. It should not re-appear on next page load.
+  * This update of the data model should happen in a `deleteFromStorage` method that is defined in the `Idea` class.
+  * How the dom is updated is up to you.
+
```

Comment:
What do you mean by "How the dom is updated is up to them?"

Reply:
Trying to decouple the data model from how it is shown on the dom. So maybe there is a better way to phrase that?
The `deleteFromStorage` method should only delete the idea from storage. There should be a another function that removes it from the dom.